db/migrate/20170703140743_fix_filters_with_multiple_resource_types.rb

```diff
@@ -0,0 +1,29 @@
+class FixFiltersWithMultipleResourceTypes < ActiveRecord::Migration
+  def change
+    Filter.unscoped.all.each do |filter|
+      next if filter.valid?
+
+      if filter.errors[:permissions] == ['Permissions must be of same resource type']
+        # If the filter was created long time ago, it may have had several
+        # resources of the same type. To fix it, let's split it into different
+        # filters each with its own resource type.
+        filter.permissions.group_by(&:resource_type).each do |resource_type, permissions|
```

Comment:
It's an N+1 query. I don't know how many filters should be present in one system, but if there are a lot of them this can cause a really slow migration.

Reply:
We are talking about dozens of them.